### PR TITLE
Remove GH workflow and actions to generate documentation

### PR DESCRIPTION
**Pull request by aulorbe**
## Problem

We tried a variety of different solutions for pushing static documentation to our GH pages site (see `sdk-docs` repo for more information), but the rendering was very bad. 

Related PRs: 
- https://github.com/pinecone-io/go-pinecone/pull/41
- https://github.com/pinecone-io/go-pinecone/pull/42
- https://github.com/pinecone-io/go-pinecone/pull/43
- https://github.com/pinecone-io/go-pinecone/pull/44
- https://github.com/pinecone-io/go-pinecone/pull/45
- https://github.com/pinecone-io/go-pinecone/pull/46

## Solution

- Simply use the default hosting site of pkg.go.dev, and include instructions for visiting the documentation on `main` vs  the latest release-version's documentation.
- Changed the documentation badge to point to latest documentation on `main` instead of latest versioned-release's documentation.
- Remove GH workflow and action files

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - name: Setup Go
-        uses: actions/setup-go@v4
+        uses: actions/setup-go@v5
```
Comment (aulorbe) [inline .github/workflows/ci.yaml:13]: Just updated this while I was in there -- it's the latest action

File: README.md [hunk 2]
```diff
@@ -5,10 +5,15 @@
...
 
-[![Go Reference](https://pkg.go.dev/badge/github.com/pinecone-io/go-pinecone.svg)](https://pkg.go.dev/github.com/pinecone-io/go-pinecone)
+[![Go Reference](https://pkg.go.dev/badge/github.com/pinecone-io/go-pinecone.svg)](https://pkg.go.dev/github.com/pinecone-io/go-pinecone@main/pinecone)
 
 Official Pinecone Go Client
 
+## Documentation
+To see the latest documentation on `main`, visit https://pkg.go.dev/github.com/pinecone-io/go-pinecone@main/pinecone.
+
+To see the latest versioned-release's documentation, visit https://pkg.go.dev/github.com/pinecone-io/go-pinecone/pinecone.
```
Comment (aulorbe) [inline README.md:15]: This auto-resolves to the latest release

File: README.md [hunk 3]
```diff
@@ -5,10 +5,15 @@
 
 # go-pinecone
 
-[![Go Reference](https://pkg.go.dev/badge/github.com/pinecone-io/go-pinecone.svg)](https://pkg.go.dev/github.com/pinecone-io/go-pinecone)
+[![Go Reference](https://pkg.go.dev/badge/github.com/pinecone-io/go-pinecone.svg)](https://pkg.go.dev/github.com/pinecone-io/go-pinecone@main/pinecone)
```
Comment (aulorbe) [inline README.md:8]: `@main` here points at the `main` branch

**Review by austin-denoble — APPROVED** [review]
Really appreciate the diligence here, thank you Audrey!

**Pull request merged**